annotations/quadriga.py

```diff
@@ -275,3 +257,218 @@ def parse_response(raw_response):
+            if key not in node_mapping:
+                node_id = get_node_id()
+                nodes[node_id] = build_concept_node(subj, user, relationset.occursIn.created, relationset.occursIn.uri)
+                node_mapping[key] = node_id
+            # Retrieve the node id for the subject from the mapping.
+            subj_node_id = node_mapping[key]
+        # Add an edge linking the current relation node to the subject node.
+        edges.append({"source": rel_node_id, "relation": "subject", "target": subj_node_id})
+        
+        # -------------------------------
+        # Process the predicate of the relation.
+        # -------------------------------
+        pred = relation.predicate
+        key = f"app-{pred.id}-{pred.created.isoformat()}"
+        if key not in node_mapping:
+            node_id = get_node_id()
+            nodes[node_id] = build_concept_node(pred, user, relationset.occursIn.created, relationset.occursIn.uri)
+            node_mapping[key] = node_id
+        pred_node_id = node_mapping[key]
+        # Add an edge linking the current relation node to the predicate node.
+        edges.append({"source": rel_node_id, "relation": "predicate", "target": pred_node_id})
+        
+        # -------------------------------
+        # Process the object of the relation.
+        # -------------------------------
+        obj = relation.object_content_object
+        if isinstance(obj, Relation):
+            # If the object is itself a relation, process it recursively.
+            obj_node_id = process_relation(obj)
+        else:
+            key = f"app-{obj.id}-{obj.created.isoformat()}"
+            if key not in node_mapping:
+                node_id = get_node_id()
+                nodes[node_id] = build_concept_node(obj, user, relationset.occursIn.created, relationset.occursIn.uri)
+                node_mapping[key] = node_id
+            obj_node_id = node_mapping[key]
+        # Add an edge linking the current relation node to the object node.
+        edges.append({"source": rel_node_id, "relation": "object", "target": obj_node_id})
+        
+        # Return the node id for this processed relation event.
+        return rel_node_id
+
+    # Process the top-level (root) relation.
+    top_relation = relationset.root
+    process_relation(top_relation)
+    
+    # default mapping from the top-level relation's roles.
+    top_subj = top_relation.source_content_object
+    top_pred = top_relation.predicate
+    top_obj = top_relation.object_content_object
+    default_mapping = {
+        "subject": {"type": "REF", "reference": node_mapping.get(f"app-{top_subj.id}-{top_subj.created.isoformat()}", "")},
+        "predicate": {"type": "URI", "uri": top_pred.interpretation.label, "label": top_pred.interpretation.label},
+        "object": {"type": "REF", "reference": node_mapping.get(f"app-{top_obj.id}-{top_obj.created.isoformat()}", "")}
+    }
```

Comment:
not sure I understand what this is doing, but the default relation would need to be configured in the template that what used to create a relation.